pyiron_base/job/factory.py

```diff
@@ -66,15 +76,42 @@ def wrapper(job_name, delete_existing_job=False, delete_aborted_job=False) -> Ty
         else:
             raise AttributeError("no job class named '{}' defined".format(name))
 
-    @property
-    @abstractmethod
-    def _job_class_dict(self):
-        pass
+    def __call__(
+            self,
+            job_type: str,
+            job_name: str,
+            delete_existing_job: bool = False,
+            delete_aborted_job: bool = False
+    ) -> GenericJob:
+        """
+        Create a job.
+
+        Args:
+            job_type (str): The job class to be instantiated.
+            job_name (str): name of the job
+            delete_existing_job (bool): delete an existing job - default false
+            delete_aborted_job (bool): delete an existing and aborted job - default false
+
+        Returns:
+            GenericJob: job object depending on the job_type selected
+        """
+        job_name = job_name.replace(".", "_")
+        job = JobType(
+            class_name=self._job_class_dict[job_type],  # Pass the class directly, JobType can handle that
```

Comment:
For devs, a simple 
```suggestion
            class_name=job_type,  # Pass the class directly, JobType can handle that
```
would be more powerful, since this way one may define an own class in the notebook (or somewhere else) and use this class. If `job_type` is one of the job types in the dictionary, `JobType` will use the same class you are providing here.

Reply:
👍 I have some worries that it will break, but I'm going commit it and build some tests -- if I can't get them to pass then I'll just revert.